docs/UserGuide.md

```diff
@@ -67,15 +67,20 @@ Format: `clear`
 
 Deletes the specified person from the HippoCamus' Contact List.
 
-Format: `delete INDEX [INDEX…]`
-
-* Deletes the person at the specified `INDEX`.
-* The index refers to the index number shown in the displayed person list (without sorting).
-* The index must be a positive integer valid in the list.
+Format: `delete {INDEX [INDEX]... | -t TAG [-t TAG]...}`
+* If provided with index(es)
+  * Deletes the person at the specified `INDEX`.
+  * All indexes refers to the index number shown in the displayed person list (without sorting).
+  * All indexes must be a positive integer valid in the list.
+* If provided with tags
+  * Delete every person with the given tag.
+  * If the person is tagged by another tag, only person's tag is removed.
 
 Examples:
 * `delete 3` deletes contact at 3rd index.
 * `delete 3 4 5` deletes contacts at 3rd, 4th and 5th index.
+* `delete -t colleague` deletes contact with tag "colleague".
+* `delete -t colleague -t cs2103` deletes contacts with tag "colleague" and contacts with tag "cs2103"
```

Comment:
```suggestion
* `delete -t colleague -t cs2103` deletes the "colleague" tag and "cs2103" tag in contacts that contains these tags. If "colleague" and "cs2103" are the only tags that that contact contains, the contact will be deleted.
```

Reply:
This is a little confusing. By deleting the contact when no more tags exists, it suggests that contacts cannot have no tags. But what about a workflow where user wants to delete all tags, then assign new tags from scratch?

Should the `delete -t` command just deal with tag deletion alone, and not touch contact deletion at all?

Reply:
Does this mean we should create another command just to handle contact deletion and this `delete -t` command only deals with tag deletion? Basically similar to our `add`, `find` and `delete` commands that only deal with contacts while `tags` command only deals with tags.

Reply:
I think using the current `add`, `find`, `delete` for contact processing is a good idea, like what you mentioned. That's essentially the existing commit: "deletes all contacts with at least one of colleague or cs2103 tags".

It's the proposed change that I don't quite catch, i.e. "delete tags in contact + delete contact if there is an exact tag match".

PS: On this note, going by your suggestion to use `tags` exclusively for tag, we can do tag deletion using the same command, e.g. `tags -d -t colleague -t cs2103`. Sounds like a pretty self-consistent idea.

Reply:
Maybe is just the phrasing confusing. So in the latest commit I phrased the main info as
* If provided with tags
  * Delete every person who is tagged with the specified tag.
  * If the person is tagged with another tag, only the specified tag will be removed. The contact will not be deleted.
 
 Then for these examples, I left it as
" `delete -t colleague -t cs2103` deletes contacts with tag "colleague" and contacts with tag "cs2103" "
Cos the happy path is to delete the contact. Then ONLY IF it is tagged with another tag then the contact is not deleted.